### ISSUE TITLE
Search page crashes with special characters

### ISSUE THREAD

**Issue by Betree** [opencollective/opencollective#2969]
**Describe the bug**
Search page seems to crash with certain characters like `#`.

**To Reproduce**
- Go to https://opencollective.com/search?q=f%23 
- **Or** put `f#` in the search bar.

**Expected behavior**
- Search should work
- `#` should be used in the search, so searching for `c#` should return collectives tagged with that string

**Screenshots**
![image](https://user-images.githubusercontent.com/1556356/76197830-2db1ec80-61ed-11ea-9c97-5d467e2a7d98.png)

**Additional context**
Reported on https://opencollective.freshdesk.com/a/tickets/4899

**Comment by bolariin** [comment]
I am having some trouble recreating this locally. I have added the Algolia keys but I still don't get the error message. Is there any other search related thing that I might be missing?

**Comment by Betree** [comment]
@bolariin Algolia is configured on the API side, have you edited the `.env` in `opencollective-api`?

A good test would be to add a `console.log` in https://github.com/opencollective/opencollective-api/blob/9d88ee8b596629e066dd94ea052701d0c7f175f5/server/graphql/v1/queries.js#L1443 to see what `Algolia.isAvailable` returns.

FYI, when Algolia is not configured we default to a custom search function that uses good old SQL.

**Comment by bolariin** [comment]
@Betree Thanks for helping me with this but it looks like I can't recreate the bug.

> have you edited the `.env` in `opencollective-api`?

I had added the Algolia keys to `default.json` in the `config` directory but I created a `.env` but it had the same result. 

Using `console.log`, it looks like Algolia is available. I tried the search with and without the Algolia keys.

**Comment by Betree** [comment]
I did investigate a bit more and through the `search` is what triggers it, the problem actually comes from `collective > stats > totalAmountSpent` on [this collective](https://opencollective.com/fnrs). I'm able to reproduce with this GQL query:

```gql
{
  Collective(slug: "fnrs") {
    stats {
      totalAmountSpent
    }
  }
}
```

By digging a bit more, it results that `getFxRate` is the source of the crash, and that's because the `collective.currency` is `€` (while it should be `EUR`). I'm not sure how it happened, but I've fixed the two entries that we had in the DB (49466 + 2849), and now https://opencollective.com/search?q=f%23 works again :tada: 

Thanks for your help narrowing down the problem @bolariin.
